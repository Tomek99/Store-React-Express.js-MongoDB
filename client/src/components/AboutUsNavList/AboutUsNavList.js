import React from "react";
import NavListBtn from "./NavListBtn/NavListBtn";
import styles from "./AboutUsNavList.module.scss";
import PropTypes from "prop-types";

function AboutUsNavList({ switchTab, tabNumber }) {
  const navBarList = [
    { name: "About Us", number: 0 },
    { name: "Our Team", number: 1 },
    { name: "FAQ", number: 2 },
  ];
  return (
    <nav className={styles.AboutUsNavList}>
      <ul className={styles.navListElements}>
        {navBarList.map((item, index) => (
          <NavListBtn
            key={index}
            item={item}
            switchTab={switchTab}
            tabNumber={tabNumber}
          />
        ))}
      </ul>
    </nav>
  );
}

AboutUsNavList.propTypes = {
  switchTab: PropTypes.func,
  tabNumber: PropTypes.number,
};
export default AboutUsNavList;
