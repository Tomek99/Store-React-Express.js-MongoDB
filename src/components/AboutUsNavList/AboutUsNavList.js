import React from "react";
import NavListElement from "../NavListElement/NavListElement";
import styles from "./AboutUsNavList.module.scss";

function AboutUsNavList() {
  const navBarList = [
    { name: "about us", path: "about-coffee-shop" },
    { name: "our team", path: "our-team" },
    { name: "FAQ", path: "faq" },
  ];
  return (
    <nav className={styles.AboutUsNavList}>
      <ul className={styles.navListElements}>
        {navBarList.map((item) => (
          <NavListElement
            isLink={false}
            key={item.name}
            name={item.name}
            path={item.path}
            activeStyle={styles.activeStyle}
          />
        ))}
      </ul>
    </nav>
  );
}

export default AboutUsNavList;
