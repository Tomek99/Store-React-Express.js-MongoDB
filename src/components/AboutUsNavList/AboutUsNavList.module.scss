.AboutUsNavList {
    display: flex;
    justify-content: center;

    .navListElements {
        display: flex;
        justify-content: space-evenly;
        flex-wrap: wrap;
        align-items: center;
        list-style: none;
        max-width: 600px;
        flex-basis: 100%;
        font-size: 2rem;
        margin: 10px;
        a {
            color: #fff;
        }
        li {
            position: relative;
        }
        a:hover {
            color: var(--main-color);
        }

        .activeStyle {
            color: var(--main-color);
        }
        .activeStyle::after {
            color: var(--main-color);
            display: block;
            content: "";
            width: 100%;
            height: 1px;
            background: var(--main-color);
            position: absolute;
            bottom: 0;
            left: 0;
        }
    }
}

@media (max-width: 500px) {
    .navListElements {
        max-width: 300px !important;
    }
}
