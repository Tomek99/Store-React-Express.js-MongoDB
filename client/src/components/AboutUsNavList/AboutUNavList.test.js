import { render, screen } from "@testing-library/react";
import user from "@testing-library/user-event";
import AboutUsNavList from "./AboutUsNavList";

test("Should render AbousUsNavList component", async () => {
  const { container } = render(<AboutUsNavList />);

  expect(container).toMatchSnapshot();
});

test("First button should be triggered", async () => {
  const mock = jest.fn();

  render(<AboutUsNavList switchTab={mock} />);

  const element = screen.getByRole("button", {
    name: /about us/i,
  });

  await user.click(element);

  expect(mock).toHaveBeenCalled();
  expect(mock).toHaveBeenCalledWith(0);
});

const pause = () => new Promise((resolve) => setTimeout(resolve, 100));
