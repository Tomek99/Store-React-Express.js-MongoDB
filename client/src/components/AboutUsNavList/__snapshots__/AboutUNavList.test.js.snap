// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AbousUsNavList component 1`] = `
<div>
  <nav
    class="AboutUsNavList"
  >
    <ul
      class="navListElements"
    >
      <li>
        <button>
          About Us
        </button>
      </li>
      <li>
        <button>
          Our Team
        </button>
      </li>
      <li>
        <button>
          FAQ
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
